File: CSharpPractice/LeetcodePractice/Leetcode189.cs
public class Solution189 {
    private void Reverse(int[] nums,int left,int right)
    {
        while(left<right)
        {
            int temp = nums[left];
            nums[left] =nums[right];
            nums[right] = temp;
            left++;
            right--;
        }

        return;
    }
    public void Rotate(int[] nums, int k) 
    {
        int n= nums.Length;
        k = k%n;
        Reverse(nums,0,n-1);
        Reverse(nums,0,k-1);
        Reverse(nums,k,n-1);
        return;
    }
}

File: JavaScriptPractice/Leetcode189.js
function reverse(nums,left,right)
{
    while(left<right)
    {
        let temp= nums[left];
        nums[left] = nums[right];
        nums[right] = temp;
        left++;
        right--;
    }
    return;

};
var rotate = function(nums, k) 
{   
    let n = nums.length;
    k = k%n;
    reverse(nums,0,n-1);
    reverse(nums,0,k-1);
    reverse(nums,k,n-1);
    return;
};